credentialdigger/client.py

```diff
@@ -1031,28 +1079,29 @@ def update_similar_snippets(self,
             Values lesser than 0.94 do not generally imply any relevant
             amount of similarity between snippets, and should
             therefore not be used.
-
         Returns
         -------
         int
             The number of similar snippets found and updated
         """
 
         discoveries = self.get_discoveries(repo_url, file_name)
-        model = build_embedding_model()
-        """ Compute target snippet embedding """
-        target_snippet_embedding = compute_snippet_embedding(target_snippet,
-                                                             model)
+        # Compute target snippet embedding
+        target_embedding = self.get_embedding(snippet=target_snippet)
         n_updated_snippets = 0
-        for d in discoveries:
-            if d['state'] == 'new':
-                """ Compute snippet embedding """
-                snippet_embedding = compute_snippet_embedding(d['snippet'],
-                                                              model)
-                """ Compute similarity of target snippet and snippet """
-                similarity = compute_similarity(target_snippet_embedding,
-                                                snippet_embedding)
-                if similarity > threshold:
-                    n_updated_snippets += 1
-                    self.update_discovery(d['id'], state)
-        return n_updated_snippets
+        if target_embedding:
+            for d in discoveries:
+                if (
+                    d['state'] != state
+                    and self.get_embedding(discovery_id=d['id'])
+                ):
+                    # Compute similarity of target snippet and snippet
```

Comment:
this comment is a little confusing: didn't we calculate the embedding of the target snippet before? 
Indeed, here you're using `target_embedding` and not `target_snippet`